src/java.base/unix/native/libjli/java_md.c

```diff
@@ -375,12 +375,33 @@ CreateExecutionEnvironment(int *pargc, char ***pargv,
          *     o          $JVMPATH (directory portion only)
          *     o          $JRE/lib
          *     o          $JRE/../lib
-         *     o          ZLIBNX_PATH (for AIX P9 or newer systems with NX)
+         *     o          ZLIBNX_PATH (for AIX P9 or newer systems with NX, unless -XX:-UseZlibNX is set)
          *
          * followed by the user's previous effective LD_LIBRARY_PATH, if
          * any.
          */
 
+#ifdef AIX
+        int argc = *pargc - 1; // skip the launcher name
+        char **argv = *pargv + 1;
+        char *arg = NULL;
+        jboolean useZlibNX = JNI_TRUE;
+        while (argc > 0 && *(arg = *argv) == '-') {
+            if (JLI_StrCmp(arg, "-XX:+UseZlibNX") == 0) {
+                useZlibNX = JNI_TRUE;
+            } else if (JLI_StrCmp(arg, "-XX:-UseZlibNX") == 0) {
+                useZlibNX = JNI_FALSE;
+            }
+            argc--;
+            argv++;
+        }
+        useZlibNX = useZlibNX && power_9_andup() && power_nx_gzip();
+        if (JLI_IsTraceLauncher()) {
+            printf("Add " ZLIBNX_PATH " to the LIBPATH: %s P9+ %s NX %s\n", useZlibNX ? "TRUE" : "FALSE",
+                power_9_andup() ? "TRUE" : "FALSE", power_nx_gzip() ? "TRUE" : "FALSE");
```

Comment:
Can you split this to make it easier to read?
```c
            printf("Add " ZLIBNX_PATH " to the LIBPATH: %s P9+ %s NX %s\n",
                useZlibNX ? "TRUE" : "FALSE",
                power_9_andup() ? "TRUE" : "FALSE",
                power_nx_gzip() ? "TRUE" : "FALSE");
```

Reply:
Updated for both comments.